### PR TITLE
Dropdown list (Mobile): "Max width" example and implementation

### PR DESCRIPTION
**Pull request by grzegorz-czajkowski**
Two commits:
 *  UIComponents (Mobile): add "Max width" dropdown example
 *  Dropdown list (Mobile): fit to viewport

### CODE REVIEW

File: src/css/profile/mobile/base.less [hunk 1]
```diff
@@ -76,6 +76,8 @@
 @slider-value-color: @slider-base-color;
 @slider-handler-color: @slider-base-color;
 @content-area-line-color: #d6d6d6;
+
+@dropdown-list-margin: 6 * @px_base;
 @dropdown-list-box-shadow-color: fade(@_black, 35%);
```
Comment (TomaszLukawskiSam) [inline src/css/profile/mobile/base.less]: Try to find and remove this style:
image

**Review by TomaszLukawskiSam — CHANGES_REQUESTED** [review]

File: src/css/profile/mobile/common/dropdownmenu.less [hunk 2]
```diff
@@ -179,9 +179,10 @@
 	overflow: hidden;
 	z-index: 1201;
 	min-width: 168px;
+	max-width: 100vw;
 	&.ui-dropdownmenu-options-vertical-margins {
-		margin-top: 6 * @px_base;
-		margin-bottom: 6 * @px_base;
+		margin-top: @dropdown-list-margin;
```
Comment (TomaszLukawskiSam) [inline src/css/profile/mobile/common/dropdownmenu.less]: do you need this variable for something else? If not, you can remove it.

File: src/css/profile/mobile/common/dropdownmenu.less [hunk 3]
```diff
@@ -213,6 +214,8 @@
 		list-style: none;
 		padding: 0;
 		margin: 0;
+		max-height: calc(~"100vh - "2 * @dropdown-list-margin);
```
Comment (TomaszLukawskiSam) [inline src/css/profile/mobile/common/dropdownmenu.less]: I think that better use simple `2 * 6 * @px_base` instead of additional variables.

**Review by TomaszLukawskiSam — CHANGES_REQUESTED** [review]

File: src/css/profile/mobile/base.less (hunk 1 above)
Comment (grzegorz-czajkowski) [reply src/css/profile/mobile/base.less]: Done.

File: src/css/profile/mobile/common/dropdownmenu.less (hunk 2 above)
Comment (grzegorz-czajkowski) [reply src/css/profile/mobile/common/dropdownmenu.less]: Yes, I am using it to set container max-height.

File: src/css/profile/mobile/common/dropdownmenu.less (hunk 3 above)
Comment (grzegorz-czajkowski) [reply src/css/profile/mobile/common/dropdownmenu.less]: Wouldn't be better to have variable for this? So we will have one place to change margin and after this container position and height are updated as well?

File: src/css/profile/mobile/theme-changeable/theme.color.less [hunk 4]
```diff
@@ -265,7 +265,6 @@
 //						DropdownMenu
 //***************************************************************************
 @color_dropdownmenu_list_group_stroke: B0733L1; // #[color] list stroke line color
-@color_dropdownmenu_list_item_bg_press: B042; // #[color] list item background press color
```
Comment (TomaszLukawskiSam) [inline src/css/profile/mobile/theme-changeable/theme.color.less:268]: I think I did mistake in comment.
You can remove this line and replace this variable in:
image
by `var(--ripple-color)`

File: src/css/profile/mobile/common/dropdownmenu.less [hunk 5]
```diff
@@ -213,6 +214,8 @@
...
+		overflow-y: auto;
 		background-color: var(--dropdown-menu-options-background);
 		border-radius: 26 * @px_base;
 		box-shadow: inset 0 -3 * @px_base 0 0 @dropdown-list-box-shadow-color;
```
Comment (TomaszLukawskiSam) [inline src/css/profile/mobile/common/dropdownmenu.less]: My comment is out of scope this change but I think that the `box-shadow` is different as in guideline. `0px 0px 3px 0px rgba(0,0,0,35)`

File: src/css/profile/mobile/theme-changeable/theme.color.less (hunk 4 above)
Comment (TomaszLukawskiSam) [reply src/css/profile/mobile/theme-changeable/theme.color.less:268]: But I don't sure if the press effect is on dropdown menu items.

File: src/css/profile/mobile/common/dropdownmenu.less (hunk 3 above)
Comment (TomaszLukawskiSam) [reply src/css/profile/mobile/common/dropdownmenu.less]: I understand you point of view but you need search this value across the project now. In my opinion, if you want to leave this variable you should define it in this file eg. on beginning.

Comment (grzegorz-czajkowski) [reply src/css/profile/mobile/common/dropdownmenu.less]: Ok, moved to the beginning of this file.

File: src/css/profile/mobile/common/dropdownmenu.less (hunk 5 above)
Comment (grzegorz-czajkowski) [reply src/css/profile/mobile/common/dropdownmenu.less]: css box-shadow has different kind of two first parameters (not including inset option). They are (x,y) positions of shadow box not size of border in px. That's why we can not apply them one by one.
I assume the guide intention of
`Box-shadow: 0px 0px 3px 0px rgba(0,0,0,35)`
is to add shadow on each edge, probably starting clockwise.

File: src/css/profile/mobile/theme-changeable/theme.color.less (hunk 4 above)
Comment (grzegorz-czajkowski) [reply src/css/profile/mobile/theme-changeable/theme.color.less:268]: Yes, @lslachciak noticed that ripple-effect is missing. However, proposed change is not sufficient to have ripple-effect working - it does not disappear. There is a separate issue https://github.com/Samsung/TAU/issues/832 for this and let's take care of this separately.

File: src/css/profile/mobile/common/dropdownmenu.less (hunk 5 above)
Comment (TomaszLukawskiSam) [reply src/css/profile/mobile/common/dropdownmenu.less]: https://developer.mozilla.org/pl/docs/Web/CSS/box-shadow

Comment (grzegorz-czajkowski) [reply src/css/profile/mobile/common/dropdownmenu.less]: Ok will try this.

Comment (grzegorz-czajkowski) [reply src/css/profile/mobile/common/dropdownmenu.less]: I applied this change, only rounded borders are having shadow.. Added separate commit fixing this. Thank you.

**Review by TomaszLukawskiSam — APPROVED** [review]
Ok. Looks good to me.

**Pull request merged**